.github/workflows/ci.yaml

```
@@ -48,10 +48,30 @@ jobs:
       - name: build
         run: just build
 
+  fmt:
+    name: Format (rustfmt +nightly)
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - uses: extractions/setup-just@v1
+        with:
+          just-version: "1.23.0"
+      - uses: actions/cache@v3
```

```suggestion
      - uses: actions/cache@v4
```

Doing my best to cause conflicts wherever I can.